addons/hw_posbox_homepage/controllers/homepage.py

```diff
@@ -163,24 +164,36 @@ def generate_password(self):
             'password': helpers.generate_password(),
         })
 
-    @http.route('/hw_posbox_homepage/upgrade', auth="none", type="http", cors='*')
-    def upgrade_iotbox(self):
-        commit = subprocess.check_output(
-            ["git", "--work-tree=/home/pi/odoo/", "--git-dir=/home/pi/odoo/.git", "log", "-1"]).decode('utf-8').replace("\n", "<br/>")
-        flashToVersion = helpers.check_image()
-        actualVersion = helpers.get_version()
-
-        if flashToVersion:
-            flashToVersion = '%s.%s' % (flashToVersion.get(
-                'major', ''), flashToVersion.get('minor', ''))
+    @http.route('/hw_posbox_homepage/version_info', auth="none", type="http", cors='*')
+    def get_version_info(self):
+        git = ["git", "--work-tree=/home/pi/odoo/", "--git-dir=/home/pi/odoo/.git"]
+        # Check branch name and last commit hash on IoT Box
+        current_commit = subprocess.run([*git, "rev-parse", "HEAD"], stdout=subprocess.PIPE, check=False)
+        current_branch = subprocess.run(
+            [*git, "rev-parse", "--abbrev-ref", "HEAD"], stdout=subprocess.PIPE, check=False
+        )
+        if current_commit.returncode != 0 and current_branch.returncode != 0:
+            return
+        current_commit = current_commit.stdout.decode().strip()
+        current_branch = current_branch.stdout.decode().strip()
+
+        # Fetch latest commit hash from GitHub API
+        try:
+            response = requests.get(f"https://api.github.com/repos/odoo/odoo/commits/{current_branch}")
```

Comment:
I'm pretty surprised that you used this approach, is is not better to run a git command locally to determine this information ?